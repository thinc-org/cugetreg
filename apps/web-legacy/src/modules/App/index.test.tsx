import { shallow } from 'enzyme'
import { afterAll, beforeEach, describe, expect, it, vi } from 'vitest'

describe('App module', () => {
  const mockRouter = vi.fn()
  const mockUseApp = vi.fn()
  const mockPageProps = { test: 'test' }
  const mockComponent = () => <div></div>
  const mockClient = vi.fn()
  const mockUseSaveStudyProgram = vi.fn()

  vi.doMock('./hooks/useApp', () => ({ useApp: mockUseApp }))
  vi.doMock('@web/common/hooks/useCourseGroup', () => ({
    useSaveStudyProgram: mockUseSaveStudyProgram,
  }))
  vi.doMock('@web/services/apollo', () => ({ client: mockClient }))

  const DEFAULT_ENV = process.env

  beforeEach(() => {
    vi.resetModules()
    vi.clearAllMocks()
    process.env = { ...DEFAULT_ENV }
  })

  afterAll(() => {
    process.env = DEFAULT_ENV
  })

  it.each`
    forceDark | environment
    ${true}   | ${'dev'}
    ${true}   | ${'beta'}
    ${true}   | ${'production'}
    ${false}  | ${'dev'}
    ${false}  | ${'beta'}
    ${false}  | ${'production'}
  `(
    'Should app component render preperly when forceDark=$forceDark',
    async ({ forceDark, environment }) => {
      process.env.NEXT_PUBLIC_ENVIRONMENT = environment

      const { App } = await import('.')
      const { AppProvider } = await import('@web/modules/App/context')
      const { DefaultSeo } = await import('next-seo')
      const wrapper = shallow(
        <App
          forceDark={forceDark}
          Component={mockComponent}
          router={mockRouter as any}
          pageProps={mockPageProps}
        />
      )
      expect(mockUseApp).toBeCalledWith(mockRouter)
      expect(mockUseSaveStudyProgram).toBeCalledTimes(1)
      expect(wrapper.find(mockComponent).props()).toEqual(mockPageProps)
      expect(wrapper.find(AppProvider).prop('forceDark')).toBe(forceDark)
      expect(wrapper.find(DefaultSeo).prop('dangerouslySetAllPagesToNoFollow')).toBe(
        environment !== 'production'
      )
      expect(wrapper.find(DefaultSeo).prop('dangerouslySetAllPagesToNoIndex')).toBe(
        environment !== 'production'
      )
      expect(wrapper).toMatchSnapshot()
    }
  )
})
