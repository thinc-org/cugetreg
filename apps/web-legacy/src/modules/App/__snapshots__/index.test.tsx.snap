// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App module > Should app component render preperly when forceDark=false 1`] = `
<AppProvider
  emotionCache={
    {
      "insert": [Function],
      "inserted": {},
      "key": "css",
      "nonce": undefined,
      "registered": {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             .go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "insertionPoint": undefined,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": [],
      },
    }
  }
  forceDark={false}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        [
          {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={true}
      dangerouslySetAllPagesToNoIndex={true}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": [
            {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:4200/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:4200",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
    <Memo(GoogleOneTapPrompt) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module > Should app component render preperly when forceDark=false 2`] = `
<AppProvider
  emotionCache={
    {
      "insert": [Function],
      "inserted": {},
      "key": "css",
      "nonce": undefined,
      "registered": {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             .go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "insertionPoint": undefined,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": [],
      },
    }
  }
  forceDark={false}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        [
          {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={true}
      dangerouslySetAllPagesToNoIndex={true}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": [
            {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:4200/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:4200",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
    <Memo(GoogleOneTapPrompt) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module > Should app component render preperly when forceDark=false 3`] = `
<AppProvider
  emotionCache={
    {
      "insert": [Function],
      "inserted": {},
      "key": "css",
      "nonce": undefined,
      "registered": {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             .go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "insertionPoint": undefined,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": [],
      },
    }
  }
  forceDark={false}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        [
          {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={false}
      dangerouslySetAllPagesToNoIndex={false}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": [
            {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:4200/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:4200",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
    <Memo(GoogleOneTapPrompt) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module > Should app component render preperly when forceDark=true 1`] = `
<AppProvider
  emotionCache={
    {
      "insert": [Function],
      "inserted": {},
      "key": "css",
      "nonce": undefined,
      "registered": {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             .go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "insertionPoint": undefined,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": [],
      },
    }
  }
  forceDark={true}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        [
          {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={true}
      dangerouslySetAllPagesToNoIndex={true}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": [
            {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:4200/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:4200",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
    <Memo(GoogleOneTapPrompt) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module > Should app component render preperly when forceDark=true 2`] = `
<AppProvider
  emotionCache={
    {
      "insert": [Function],
      "inserted": {},
      "key": "css",
      "nonce": undefined,
      "registered": {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             .go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "insertionPoint": undefined,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": [],
      },
    }
  }
  forceDark={true}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        [
          {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={true}
      dangerouslySetAllPagesToNoIndex={true}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": [
            {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:4200/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:4200",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
    <Memo(GoogleOneTapPrompt) />
  </TrackPageChange>
</AppProvider>
`;

exports[`App module > Should app component render preperly when forceDark=true 3`] = `
<AppProvider
  emotionCache={
    {
      "insert": [Function],
      "inserted": {},
      "key": "css",
      "nonce": undefined,
      "registered": {},
      "sheet": StyleSheet {
        "_insertTag": [Function],
        "before": null,
        "container": <head>
          <style
            id="_goober"
          >
             @keyframes go2264125279{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go3020080000{from{transform:scale(0);opacity:0;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go463499852{from{transform:scale(0) rotate(90deg);opacity:0;}to{transform:scale(1) rotate(90deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1268368563{from{transform:rotate(0deg);}to{transform:rotate(360deg);}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go1310225428{from{transform:scale(0) rotate(45deg);opacity:0;}to{transform:scale(1) rotate(45deg);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go651618207{0%{height:0;width:0;opacity:0;}40%{height:0;width:6px;opacity:1;}100%{opacity:1;height:10px;}}
          </style>
          <style
            id="_goober"
          >
             @keyframes go901347462{from{transform:scale(0.6);opacity:0.4;}to{transform:scale(1);opacity:1;}}
          </style>
          <style
            id="_goober"
          >
             .go4109123758{z-index:9999;}.go4109123758 &gt; *{pointer-events:auto;}
          </style>
        </head>,
        "ctr": 0,
        "insertionPoint": undefined,
        "isSpeedy": false,
        "key": "css",
        "nonce": undefined,
        "prepend": undefined,
        "tags": [],
      },
    }
  }
  forceDark={true}
>
  <TrackPageChange>
    <DefaultSeo
      additionalMetaTags={
        [
          {
            "content": "cu,chula,จุฬา,reg,getreg,regchula,ตารางเรียน,ตารางสอน,gened,CU Get Reg,ลงทะเบียนเรียน,อำนวยความสะดวก",
            "property": "keywords",
          },
          {
            "content": "CU Get Reg",
            "name": "application-name",
          },
        ]
      }
      dangerouslySetAllPagesToNoFollow={false}
      dangerouslySetAllPagesToNoIndex={false}
      defaultTitle="CU Get Reg"
      description="เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน"
      openGraph={
        {
          "description": "เว็บไซต์เพื่ออำนวยความสะดวกในการลงทะเบียนเรียนให้กับนิสิตจุฬาฯ ทุกคน",
          "images": [
            {
              "alt": "CU Get Reg",
              "height": 600,
              "url": "http://localhost:4200/cover.jpg",
              "width": 800,
            },
          ],
          "title": "CU Get Reg",
          "type": "website",
          "url": "http://localhost:4200",
        }
      }
      titleTemplate="%s | CU Get Reg"
    />
    <LoadingProgress />
    <CssBaseline />
    <TopBar />
    <ForwardRef(Container)>
      <ErrorBoundary>
        <mockComponent
          test="test"
        />
      </ErrorBoundary>
    </ForwardRef(Container)>
    <Footer />
    <ShoppingCartModal />
    <CustomToaster />
    <ForwardRef(LoadableComponent) />
    <Memo(GoogleOneTapPrompt) />
  </TrackPageChange>
</AppProvider>
`;
